backend/corpora/common/submission.py

```diff
@@ -0,0 +1,115 @@
+import os
+import re
+from typing import Optional, Tuple
+import logging
+from sqlalchemy.orm import Session
+
+from backend.corpora.common.entities import Dataset, Collection
+from backend.corpora.common.utils.corpora_constants import CorporaConstants
+from backend.corpora.common.utils.exceptions import CorporaException, NonExistentCollectionException
+from backend.corpora.common.utils.regex import (
+    USERNAME_REGEX,
+    COLLECTION_ID_REGEX,
+    DATASET_ID_REGEX,
+    CURATOR_TAG_PREFIX_REGEX,
+    EXTENSION_REGEX,
+)
+from backend.corpora.common.upload import upload
+
+logger = logging.getLogger(__name__)
+
+REGEX = f"^{USERNAME_REGEX}/{COLLECTION_ID_REGEX}/({DATASET_ID_REGEX}|{CURATOR_TAG_PREFIX_REGEX})\\.{EXTENSION_REGEX}$"
```

Comment:
I think we should put this in a common file, have a single source, given that it is already [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/9af989384aaf660182036996655f817186f5978d/backend/corpora/dataset_submissions/app.py#L22)